backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs

```diff
@@ -90,14 +88,43 @@ public async Task<List<UserShortInfoDto>> GetFilteredUsers(UserFilterDto userFil
             filteredUsers = filteredUsers.Where(u => u.Tags.Any(t => filter.Topics.Contains(t.Name)));
         }
         var filteredUsersList = await filteredUsers.ToListAsync();
-        return _mapper.Map<List<UserShortInfoDto>>(filteredUsersList);
+        var mappedUsers = _mapper.Map<List<UserShortInfoDto>>(filteredUsersList);
+        await FillUserFriendshipStatus(mappedUsers);
+        return mappedUsers;
+    }
+
+    private async Task FillUserFriendshipStatus(List<UserShortInfoDto> users)
+    {
+        var myId = _authService.UserId;
+        var requesters = await _context.Friends.Where(f=>f.FriendshipStatus == FriendshipStatus.Pending && f.UserId == myId).ToListAsync();
+        var acceptors = await _context.Friends.Where(f => f.FriendshipStatus == FriendshipStatus.Pending && f.RequesterId == myId).ToListAsync();
+        var confirmedFriends = await _context.Friends.Where(f => f.FriendshipStatus == FriendshipStatus.Confirmed && (f.RequesterId == myId || f.UserId == myId)).ToListAsync();
+        users.ForEach(user =>
+        {
+            if (requesters.Any(requester => requester.RequesterId == user.Id))
+            {
+                user.UserFriendshipStatus = UserFriendshipStatus.Requester;
+            }
+            else if (acceptors.Any(acceptor => acceptor.UserId == user.Id))
+            {
+                user.UserFriendshipStatus = UserFriendshipStatus.Acceptor;
+            }
+            else if (confirmedFriends.Any(friend => friend.UserId == user.Id || friend.RequesterId == user.Id))
+            {
+                user.UserFriendshipStatus = UserFriendshipStatus.Friend;
+            }
+            else
+            {
+                user.UserFriendshipStatus = UserFriendshipStatus.Regular;
+            }
```

Comment:
I'm not really sure if it makes sense to show users on social page...
And 3 SQL requests is a bit too much for setting status...